src/api/authentication.js

```diff
@@ -36,108 +34,20 @@ const auditingExemptPaths = [
   /\/logs/
 ]
 
-const isUndefOrEmpty = string => string == null || string === ''
-
-async function authenticateBasic(ctx) {
-  const credentials = basicAuth(ctx)
-  if (credentials == null) {
-    // No basic auth details found
-    return null
-  }
-  const {name: email, pass: password} = credentials
-  const user = await UserModelAPI.findOne({
-    email: caseInsensitiveRegex(email)
-  })
-  if (user == null) {
-    // not authenticated - user not found
-    ctx.throw(
-      401,
-      `No user exists for ${email}, denying access to API, request originated from ${ctx.request.host}`,
-      {email}
-    )
-  }
-
-  const hash = crypto.createHash(user.passwordAlgorithm)
-  hash.update(user.passwordSalt)
-  hash.update(password)
-  if (user.passwordHash !== hash.digest('hex')) {
-    // not authenticated - password mismatch
-    ctx.throw(
-      401,
-      `Password did not match expected value, denying access to API, the request was made by ${email} from ${ctx.request.host}`,
-      {email}
-    )
-  }
-  return user
-}
-
-async function authenticateToken(ctx) {
```

Comment:
Note, the console doesn't have to support both, just the core. So no changes are necessary on the console.

Reply:
@nour-borgi we spoke about this in the architecture forum. The result of that is that we feel we need to take the safe approach which is to support the token based approach for one more version while we implement these passport changes but mark it as deprecated, then remove it in a subsequent version.

So this mean we need the following changes on this PR:
* Implement a passport auth strategy that implements the old token approach
* When this method is used add a log messages that states it is being deprecated
* Implement a migration to move existing users to the new db structure - this can be done [here as we have support for migrations](https://github.com/jembi/openhim-core-js/blob/master/src/upgradeDB.js) (this needs to be done regardless of if we keep the token auth as we need to give people a way to upgrade their db)

cc @marrouchi @michaelloosen

Reply:
Well noted. Thanks